sources/hubspot/__init__.py

```diff
@@ -38,54 +38,174 @@
+    def contacts(
+        api_key: str = api_key,
+        include_history: bool = include_history,
+        props: Sequence[str] = DEFAULT_CONTACT_PROPS,
+    ) -> Iterator[TDataItems]:
+        """Hubspot contacts resource"""
+        yield from crm_objects(
+            "contact",
+            api_key,
+            include_history,
+            props if props in (ALL, CUSTOM_ONLY) else props + global_props,  # type: ignore
+        )
+
+    @dlt.resource(name="deals", write_disposition="replace")
+    def deals(
+        api_key: str = api_key,
+        include_history: bool = include_history,
+        props: Sequence[str] = DEFAULT_DEAL_PROPS,
+    ) -> Iterator[TDataItems]:
+        """Hubspot deals resource"""
+        yield from crm_objects(
+            "deal",
+            api_key,
+            include_history,
+            props if props in (ALL, CUSTOM_ONLY) else props + global_props,  # type: ignore
+        )
+
+    @dlt.resource(name="tickets", write_disposition="replace")
+    def tickets(
+        api_key: str = api_key,
+        include_history: bool = include_history,
+        props: Sequence[str] = DEFAULT_TICKET_PROPS,
+    ) -> Iterator[TDataItems]:
+        """Hubspot tickets resource"""
+        yield from crm_objects(
+            "ticket",
+            api_key,
+            include_history,
+            props if props in (ALL, CUSTOM_ONLY) else props + global_props,  # type: ignore
+        )
+
+    @dlt.resource(name="products", write_disposition="replace")
+    def products(
+        api_key: str = api_key,
+        include_history: bool = include_history,
+        props: Sequence[str] = DEFAULT_PRODUCT_PROPS,
+    ) -> Iterator[TDataItems]:
+        """Hubspot products resource"""
+        yield from crm_objects(
+            "product",
+            api_key,
+            include_history,
+            props if props in (ALL, CUSTOM_ONLY) else props + global_props,  # type: ignore
+        )
+
+    @dlt.resource(name="quotes", write_disposition="replace")
+    def quotes(
+        api_key: str = api_key,
+        include_history: bool = include_history,
+        props: Sequence[str] = DEFAULT_QUOTE_PROPS,
+    ) -> Iterator[TDataItems]:
+        """Hubspot quotes resource"""
+        yield from crm_objects(
+            "quote",
+            api_key,
+            include_history,
+            props if props in (ALL, CUSTOM_ONLY) else props + global_props,  # type: ignore
+        )
+
+    return companies, contacts, deals, tickets, products, quotes
 
 
 def crm_objects(
     object_type: str,
     api_key: str = dlt.secrets.value,
     include_history: bool = False,
+    props: Sequence[str] = None,
 ) -> Iterator[TDataItems]:
     """Building blocks for CRM resources."""
-    props = ",".join(_get_property_names(api_key, object_type))
+    if props == ALL:
+        props = ",".join(_get_property_names(api_key, object_type))
+    elif props == CUSTOM_ONLY:
```

Comment:
this is almost good. what I wanted is to get the custom props on top of `props` passed by the user. so defines the hubspot props but custom come automatically. let's do a boolean flag to include them

Reply:
Not sure I understand. What are the use cases then?

`hubspot()` - return all custom properties
`hubspot(only_custom=False)` - return properties from the default `search` lists
`hubspot(only_custom=False, props=["prop1", "prop2"]` - return only properties set by the user
`hubspot(only_custom=False, props=ALL)` - return all the possible properties

Like that? It seems quite complex - two args to control actually the same thing, and the use cases are not very clear at the first glance.

With the current version the use cases are:

`hubspot()` - return properties from the default `search` lists
`hubspot(props=ALL)` - return all the possible properties
`hubspot(props=CUSTOM_ONLY)` - return all custom properties
`hubspot(props=["prop1", "prop2"])` - return only the requested properties

Actually, if we provide an ability to request all the custom properties, then default lists of properties don't make much sense anymore. So, as I see it, it should be like this then:

`hubspot()` - return all the custom properties
`hubpost(props=ALL)` - return all the possible properties
`hubspot(props=["prop1", "prop2"])` - return only the given properties

And those default lists of properties are not needed at all.

Reply:
O-o-oh, or maybe you want to combine all custom props to whatever the user passes into the `props` arg? Like that:

`hubspot()` - return all the custom properties
`hubspot(props=[...])` - all custom props + `[...]`?
`hubspot(props=[...], add_custom=False)` - only `[...]` is read?
`hubspot(add_custom=False)` - what to return in this case then?

Reply:
yeah! that was what I mean thx :)